### ISSUE TITLE
QC: Panel de strategies - tracker d'amelioration iterative

### ISSUE THREAD

**Issue by jsboige** [jsboige/CoursIA#29]
### Methodologie

**Philosophie iteration-2 (robustesse > performance):**
- Etendre les periodes de backtest au maximum (pas de cherry-picking)
- Walk-forward validation pour confirmer la stabilite
- Accepter un Sharpe plus bas si la strategie tient sur toutes les conditions de marche
- Pas de suroptimisation des parametres

Pour chaque strategie:
1. Creer un **research notebook** pedagogique (standalone yfinance+pandas)
2. Explorer et comprendre POURQUOI la strategie fonctionne/echoue
3. Implementer les findings dans main.py
4. Backtester sur la plus longue periode possible
5. Valider la robustesse (walk-forward, regime analysis)

### Commande

```
/qc-iterative-improve [strategy|issue#] [--iterations=3]
```

Pour reprendre apres un redemarrage: `/continue` (lit les issues GitHub).

#### BROKEN (Sharpe < 0) - A corriger
- [ ] #17 ForexCarry (Sharpe -1.80 -> **-0.654**) - FX momentum, long-only
- [ ] #18 VIX-TermStructure (Sharpe -0.97 -> **-0.27**) - SVXY, VIX<22 filter

#### NEEDS_IMPROVEMENT (Sharpe 0-0.5) - A ameliorer
- [ ] #19 MeanReversion (Sharpe -0.042 -> **0.294**) - RSI multi-asset
- [ ] #20 FuturesTrend (Sharpe 0.019 -> **0.280**) - Donchian ETFs
- [ ] #21 TurnOfMonth (Sharpe 0.127 -> **0.127**) - CEILING confirme
- [ ] #22 MomentumStrategy (Sharpe 0.216 -> **0.411**) - XLP+XLU risk-off
- [ ] #23 AllWeather (Sharpe 0.25 -> **0.365**) - GLD 20%, no SMA overlay
- [ ] #25 FamaFrench (Sharpe 0.365 -> **0.471**) - USMV risk-off
- [ ] #34 Crypto-MultiCanal (0 trades) - Copie depuis ESGF, a diagnostiquer

#### HEALTHY (Sharpe > 0.5) - Valider robustesse
- [ ] #24 OptionsIncome (Sharpe 0.288 -> **0.747**) - Covered call, VIX filter
- [ ] #26 Sector-Momentum (Sharpe 0.554) - Dual Momentum
- [ ] #32 BTC-MACD-ADX (Sharpe **1.224**) - Crypto C#, ex-ESGF
- [ ] #33 Multi-Layer-EMA (Sharpe **1.891**) - EMA Python, ex-ESGF

#### Infrastructure et documentation
- [x] #28 Update iterative improvement workflow (skill + agents)
- [x] #30 README principal: refonte pedagogique et mise a jour
- [ ] #31 Email Jared Broad: finaliser le brouillon et envoyer

### Tableau de bord

| # | Strategie | Sharpe v0 | Sharpe v1 | QC ID | Lang | Statut |
|---|-----------|-----------|-----------|-------|------|--------|
| #17 | ForexCarry | -1.80 | **-0.654** | 28657908 | Py | IMPROVED |
| #18 | VIX-TermStructure | -0.97 | **-0.27** | 28657907 | Py | IMPROVED |
| #19 | MeanReversion | -0.042 | **0.294** | 28657904 | Py | IMPROVED |
| #20 | FuturesTrend | 0.019 | **0.280** | 28657834 | Py | IMPROVED |
| #21 | TurnOfMonth | 0.127 | 0.127 | 28657905 | Py | CEILING |
| #22 | MomentumStrategy | 0.216 | **0.411** | 28657837 | Py | IMPROVED |
| #23 | AllWeather | 0.25 | **0.365** | 28657833 | Py | IMPROVED |
| #24 | OptionsIncome | 0.288 | **0.747** | 28657838 | Py | IMPROVED |
| #25 | FamaFrench | 0.365 | **0.471** | 28657910 | Py | IMPROVED |
| #26 | Sector-Momentum | 0.554 | 0.554 | 28433643 | Py | HEALTHY |
| #32 | BTC-MACD-ADX | - | **1.224** | 28418632 | C# | HEALTHY |
| #33 | Multi-Layer-EMA | - | **1.891** | 28433748 | Py | HEALTHY |
| #34 | Crypto-MultiCanal | - | 0 trades | 28679473 | Py | BROKEN |

### Patterns decouverts (iteration-1)

- **TLT risk-off detruit la valeur** en 2015-2026 (hausse taux 2022). Alternatives: XLP+XLU, USMV, cash.
- **12m simple lookback >> composite/risk-adjusted** pour la plupart des strategies.
- **Futures -> ETFs** : plus simple, pas de rollover, meilleur Sharpe.
- **SMA overlay + low-vol strategies** : tue le Sharpe (vol ~6%, risk-free ~5%).
- **VIX filter pour options** : ne vendre que quand VIX > 15.

### Cibles finales

| Metrique | Target | Acceptable |
|----------|--------|------------|
| Sharpe | > 0.5 | > 0.3 |
| Max DD | < 25% | < 35% |
| Robustesse | Walk-forward stable | Pas de collapse en bear |
| Notebook | Complet + pedagogique | Au moins exploratoire |

**Comment by jsboige** [comment]
**Stale flag** — last activity 2026-03-04 (59 days ago).

Strategy improvement tracker with 8/26 strategies improved (TurnOfMonth #21 and Sector-Momentum #26 unchanged). Is this tracker still active, or should it be closed/archived?

**Comment by jsboige** [comment]
## Update 2026-05-07

Closed issues this session:
- #28 (stale, workflow evolved)
- #19 MeanReversion: FIXED (Sharpe 0.365)
- #20 FuturesTrend: FIXED (Sharpe 0.301)
- #22 MomentumStrategy: FIXED (Sharpe 0.459)
- #626 (expired sprint tracking)

Remaining open:
- #17 ForexCarry (Sharpe -0.654, pedagogical:broken-intentional)
- #18 VIX-TermStructure (Sharpe -0.27)
- #21 TurnOfMonth (CEILING at 0.127)
- #23 AllWeather (0.365)
- #25 FamaFrench (0.471)
- #34 Crypto-MultiCanal (0 trades)

**Comment by jsboige** [comment]
## AlphaModel Framework Migration (#45) — 2026-05-07

Migrated 4 strategies to QC Algorithm Framework (AlphaModel pattern).

### Results

| Strategy | Sharpe (standalone) | Sharpe (framework) | Delta | Status |
|----------|--------------------|--------------------|-------|--------|
| EMA-Cross-Stocks | 0.872 | **1.123** | +29% | VALID |
| DualMomentumNoTLT | ~0.469 | **0.634** | +35% | VALID |
| MomentumStrategy (#22) | 0.472 | **0.478** | +1% | VALID |
| AdaptiveAssetAllocation | 0.518 | **0.343** | -34% | NEEDS FIX |

### Cloud Project IDs

- Framework-EMA-Cross-Stocks: 31244059
- Framework-DualMomentumNoTLT: 31244186
- Framework-AdaptiveAssetAllocation: 31244187
- Framework-MomentumStrategy: 31244262

### Pattern

Each strategy now has:
- `alpha_model.py`: AlphaModel subclass with `update()` and `on_securities_changed()`
- `config_framework.json`: Cloud project IDs (separate from standalone config)
- `main.py` on cloud: Uses `set_alpha()`, `InsightWeightingPortfolioConstructionModel()`, `ImmediateExecutionModel()`

### Notes

- 3/4 within +/-15% target. AdaptiveAA underperforms due to min-variance weight normalization difference.
- EMA-Cross-Stocks significantly improved — framework rebalancing is more efficient than manual `set_holdings` with 5% threshold.
- Commit: `fb9f51d9` on `feat/ml-gnn-crypto-panier`

**Comment by jsboige** [comment]
## Cycle 5 Updates (2026-05-07)

### AdaptiveAA Framework Fix

AdaptiveAA Sharpe improved from 0.343 to **0.482** (commit `90d14d76`).
Fix: extended insight period to 45 days (overlaps with next monthly emission, preventing expiry gaps).

**Updated framework results:**

| Strategy | Sharpe (standalone) | Sharpe (framework) | Delta | Status |
|----------|--------------------|--------------------|-------|--------|
| EMA-Cross-Stocks | 0.872 | **1.123** | +29% | VALID |
| DualMomentumNoTLT | ~0.469 | **0.634** | +35% | VALID |
| MomentumStrategy (#22) | 0.472 | **0.478** | +1% | VALID |
| AdaptiveAssetAllocation | 0.518 | **0.482** | -7% | VALID |

All 4 framework migrations now VALID.

### QC-Py-32 DQN NaN Bug Fix

**Bug**: Circular dependency in warmup gate — `agent.total_steps` (only incremented inside `update()`) was used as gate condition, so `update()` never ran. Loss stayed NaN, epsilon stuck at 1.0, fake Sharpe 0.657 from random buying.

**Fix**: External `total_transitions` counter (commit `2353c88c`).

**Verified results**: Sharpe **0.704** genuine, loss 0.003-0.008, epsilon decays properly, MaxDD -21%.

### QC-Py-25 RL Enrichment

4 student exercises added (commit `ad99b937`):
1. DQN hyperparameter impact comparison
2. RSI + Bollinger %B environment enrichment
3. Whipsaw penalty reward shaping
4. Early stopping tracker with patience

### BROKEN Strategy Categorization (Track G Proposal)

**BROKEN_PEDAGOGICAL** (intentional failure demonstrations):
- ForexCarry (#17): FX carry dead post-2008, teaches structural regime failure
- VIX-TermStructure (#18): SVXY terminated 2018, teaches structural product risk
- ML-HeadShoulders-CNN: Failed generalization, teaches overfitting in pattern recognition

**BROKEN_TO_FIX** (genuine bugs):
- Crypto-MultiCanal (#34): `find_envelope_line` import error (LOW effort)
- Trend-Following: Timeout 200 stocks × 11yr hourly (MEDIUM)
- Stoploss-Volatility-ML: CBOE data unavailable (MEDIUM)
- Clustering-Fundamentals-ML: Runtime error (LOW)

**NO_BACKTEST** (status unknown): 6 ML + 10 ESGF projects need initial evaluation.

**Comment by jsboige** [comment]
## po-2025 Cycle 9 Audit — Status Corrections (2026-05-08)

Diagnosed all BROKEN and NEEDS_IMPROVEMENT strategies via QC Cloud backtest data + code analysis.

### Corrections Required

| # | Strategy | Tracker Sharpe | Actual Sharpe | Status Change | Verdict |
|---|----------|---------------|---------------|---------------|---------|
| #34 | Crypto-MultiCanal | 0 trades (BROKEN) | **0.581** (73 orders) | BROKEN → **HEALTHY** | STALE tracker |
| #19 | MeanReversion | 0.294 | **0.461** IS / **0.14** OOS | Sharpe update | v5.2-fix completed |
| #17 | ForexCarry | -0.654 | -0.654 | BROKEN → **STRUCTURAL_CEILING** | FX momentum < risk-free rate post-2008 |
| #18 | VIX-TermStructure | -0.27 | **+0.05** (v4.1 BEST) | Sharpe update | SVXY -0.5x post-2018, ceiling ~0.05-0.10 |
| #20 | FuturesTrend | 0.280 | **0.301** (v3.1 BEST) | CEILING confirmed | Parametric optimization exhausted |
| #22 | MomentumStrategy | 0.411 | **0.472** (v4.0 BEST) | CEILING confirmed | Marked ARCHIVED in code |

### Detailed Findings

**Crypto-MultiCanal (#34)**: QC Cloud backtest `Diagnose 0 trades 2020-2026` shows Sharpe 0.581, CAGR 8.211%, 73 orders. The strategy was already fixed. Reclassify as HEALTHY.

**MeanReversion (#19)**: v5.2-fix (completed, no RuntimeError) = Sharpe 0.461, CAGR 9.2%, MaxDD 24.4% (2015-2024). OOS 2024-2026 = Sharpe 0.14, CAGR 9.21%. The 0.294 in tracker is stale (from v6c iteration). Best version v5.2 had Sharpe 0.810 but RuntimeError (fixed in v5.2-fix). OOS drop 0.461→0.14 suggests some overfitting to in-sample parameters.

**ForexCarry (#17)**: Research notebook tested 5 hypotheses (H1-H5). Conclusion: FX momentum G10 generates ~0.7-1.5% CAGR absolute, but risk-free rate is 2.5-5.5% on 2015-2026. Negative Sharpe is structural — the momentum doesn't compensate for opportunity cost. Menkhoff et al. (2012) document FX momentum weakening post-2008. **Negative result** — recommend documenting as pedagogical case study.

**VIX-TermStructure (#18)**: Code v4.1 achieves Sharpe +0.05. All iterations (v2.0 through v5.1) thoroughly tested. SVXY is -0.5x post-2018 VIXplosion, halving contango premium. Code docstring: "Ceiling: Sharpe ~0.05-0.10 is honest for SVXY short-vol 2015-2026". **Structural ceiling** — short-vol with -0.5x product.

**FuturesTrend (#20)**: v3.1 = Sharpe 0.301 (BEST). Research iter5 confirmed SMA parameter overfitting (OOS Sharpe -0.010). Docstring: "v3.1 is the structural ceiling for parametric optimization on this universe/period".

**MomentumStrategy (#22)**: v4.0 = Sharpe 0.472 (BEST). v5.0 through v6.2 all rejected. Code header: "Sharpe ceiling ~0.48". Marked as ARCHIVED in code.

### Actionable Items

1. **Crypto-MultiCanal**: Update tracker table — HEALTHY, Sharpe 0.581
2. **MeanReversion**: Update Sharpe to 0.461 (IS) / 0.14 (OOS). Consider OOS robustification as improvement track
3. **ForexCarry + VIX-TermStructure**: Mark as STRUCTURAL_CEILING / negative result. Pedagogical value only
4. **FuturesTrend + MomentumStrategy**: Mark as CEILING confirmed. No further iteration needed

### Remaining Improvement Candidates

Strategies where meaningful improvement may still be possible:
- **MeanReversion** (#19): OOS robustification (reduce IS/OOS gap from 0.461→0.14)
- **AllWeather** (#23): Sharpe 0.365, no recent iteration analysis
- **FamaFrench** (#25): Sharpe 0.471, near ceiling but not confirmed

**Comment by jsboige** [comment]
Closing as superseded by Epic #1409 (Ladder iterative live tracker).

**Justification (cycle 86 audit) :**

- #29 = ancien scorecard L1-L5 + #35 que #1409 absorbe et maintient en live.
- Verdicts L1-L4 locked, L5 NO BEATS documente.
- Pas d'action specifique residuelle hors de #1409 ; preserver l'archive via Closed (no merge needed).

Fermeture propre dans le cadre du nettoyage des issues >30j sans roadmap propre.
